src/Solnet.Wallet/PublicKey.cs

```diff
@@ -168,6 +168,87 @@ public override bool Equals(object obj)
+        /// Two concrete examples: <br/>
+        /// - A user wallet key must be on the curve (otherwise a user wouldn't be able to sign transactions).  <br/>
+        /// - A program derived address must NOT be on the curve.
+        /// </remarks>
+        /// <param name="key">The base58 encoded public key.</param>
+        /// <param name="validateCurve">Whether or not to validate if the public key belongs to the Ed25519 curve.</param>
+        /// <returns>Returns true if the input is a valid key, false otherwise.</returns>
+        public static bool IsValid(string key, bool validateCurve = false)
+        {
+            if(!string.IsNullOrEmpty(key))
+            {
+                try
+                {
+                    return IsValid(Encoders.Base58.DecodeData(key), validateCurve);
+                }
+                catch(Exception)
+                {
+                    return false;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if a given set of bytes forms a valid PublicKey.
+        /// </summary>
+        /// <remarks>
+        /// Any set of 32 bytes can constitute a valid solana public key. However, not all 32-byte public keys are valid Ed25519 public keys. <br/>
+        /// Two concrete examples: <br/>
+        /// - A user wallet key must be on the curve (otherwise a user wouldn't be able to sign transactions).  <br/>
+        /// - A program derived address must NOT be on the curve.
+        /// </remarks>
+        /// <param name="key">The key bytes.</param>
+        /// <param name="validateCurve">Whether or not to validate if the public key belongs to the Ed25519 curve.</param>
+        /// <returns>Returns true if the input is a valid key, false otherwise.</returns>
+        public static bool IsValid(byte[] key, bool validateCurve = false)
+        {
+            return key != null && key.Length == 32 && (!validateCurve || key.IsOnCurve());
+        }
+        
+        /// <summary>
+        /// Checks if a given set of bytes forms a valid PublicKey.
+        /// </summary>
+        /// <remarks>
+        /// Any set of 32 bytes can constitute a valid solana public key. However, not all 32-byte public keys are valid Ed25519 public keys. <br/>
+        /// Two concrete examples: <br/>
+        /// - A user wallet key must be on the curve (otherwise a user wouldn't be able to sign transactions).  <br/>
+        /// - A program derived address must NOT be on the curve.
+        /// </remarks>
+        /// <param name="key">The key bytes.</param>
+        /// <param name="validateCurve">Whether or not to validate if the public key belongs to the Ed25519 curve.</param>
+        /// <returns>Returns true if the input is a valid key, false otherwise.</returns>
+        public static bool IsValid(ReadOnlySpan<byte> key, bool validateCurve = false)
+        {
+            return key != null && key.Length == 32 && (!validateCurve || key.IsOnCurve());

```

Comment:
And here. I think since this length is specific to the `PublicKey` it can even be shortened to just `Length`?